packages/editor/src/editor/sidebar/fields/condition-builder/ConditionBuilder.tsx

```diff
@@ -0,0 +1,137 @@
+import { BasicCheckbox, BasicCollapsible, Button, Flex, IvyIcon, Tabs, TabsList, TabsTrigger } from '@axonivy/ui-components';
+import { IvyIcons } from '@axonivy/ui-icons';
+import { useState } from 'react';
+import { ConditionGroup } from './ConditionGroup';
+import type { LogicalOperator } from './Condition';
+
+interface ConditionBuilderProps {
+  onChange: (value: string) => void;
+  apply: () => void;
+}
+
+export const ConditionBuilder = ({ onChange, apply }: ConditionBuilderProps) => {
+  const [isConditionGroupEnabled, setIsConditionGroupEnabled] = useState(false);
+  const [conditionGroups, setConditionGroups] = useState<ConditionGroup[]>([
+    { conditions: [{ argument1: '', operator: '==', argument2: '', logicalOperator: 'and' }], logicalOperator: 'and' }
+  ]);
+
+  const updateCondition = (groupIndex: number, conditionIndex: number, key: string, newValue: string) => {
+    const newGroups = [...conditionGroups];
+    newGroups[groupIndex].conditions[conditionIndex] = { ...newGroups[groupIndex].conditions[conditionIndex], [key]: newValue };
+    setConditionGroups(newGroups);
```

Comment:
instead of spread the old object, you could simply use the one the set method gives you:
```suggestion
    setConditionGroups(old => {
      old[groupIndex].conditions[conditionIndex] = { ...old[groupIndex].conditions[conditionIndex], [key]: newValue };
      return old;
    });
```